### ISSUE TITLE
Calling <anything>.arguments causes a "TypeError: 'caller', 'callee', and 'arguments' properties may not be accessed on strict mode functions or the arguments objects for calls to them" error

### ISSUE THREAD

**Issue by GeoffRen** [patriksimek/vm2#404]
Minimal repro:

```ts
const sandbox = {
    testFunc: (param: { arguments: string }) => param.arguments,
};
const vm = new VM({ sandbox });
const ret = vm.run('testFunc({arguments: "test"});');
console.log(ret);
```

This will cause a `TypeError: 'caller', 'callee', and 'arguments' properties may not be accessed on strict mode functions or the arguments objects for calls to them` error to be thrown.

This started happening with 3.9.6.

**Comment by XmiliaH** [comment]
Thanks for the report. A typeof check is missing there. This only should happen for functions.

**Comment by GeoffRen** [comment]
I sent a quick pr to fix it.

**Comment by XmiliaH** [comment]
This should be fixed in 3.9.8
